src/uart.c

```diff
@@ -3,6 +3,24 @@
 
 #include <metal/uart.h>
 
+struct metal_uart *metal_uart_get_device(void)
+{
+#ifdef __METAL_DT_STDOUT_UART_HANDLE
+    return __METAL_DT_STDOUT_UART_HANDLE;
+#else
+    return NULL;
+#endif
+}
+
+int metal_uart_get_baud_by_device(void)
```

Comment:
I would say this should be `metal_uart_get_default_baud_by_device()`, if anything.

Reply:
Yes, this API name would be better and clear. If need, this API maybe should return the baud rate depends on the given uart device which you mention before.